pinot-broker/src/main/java/org/apache/pinot/broker/queryquota/HelixExternalViewBasedQueryQuotaManager.java

```diff
@@ -81,15 +81,32 @@ public class HelixExternalViewBasedQueryQuotaManager implements ClusterChangeHan
   private final AtomicInteger _lastKnownBrokerResourceVersion = new AtomicInteger(-1);
   private final Map<String, QueryQuotaEntity> _rateLimiterMap = new ConcurrentHashMap<>();
   private final Map<String, QueryQuotaEntity> _databaseRateLimiterMap = new ConcurrentHashMap<>();
+  private final Map<String, QueryQuotaEntity> _applicationRateLimiterMap = new ConcurrentHashMap<>();
   private double _defaultQpsQuotaForDatabase;
+  private double _defaultQpsQuotaForApplication;
```

Comment:
It's a cluster-level default for all applications that don't have an explicit value set.
e.g.  if we have 
default app quota = 10
'test' app quota = 20 

Then queries with 'test' applicationName option will be limited to 20 qps while 'other' or 'app_ricot' will be limited to 10 even though they don't have an explicit mapping/value set.

Reply:
Got it, thanks for the explanation. Can you add a point around this in PR description as well?
Also consider exposing a metric that tracks the number of active applications being monitored by the rate limiter map as its an unbounded map and not sure how this will be leveraged by users so having those insights would help

Reply:
> Can you add a point around this in PR description as well?

Please add that context in a javadoc as well. Usually any design decision that is worth to be in the PR description should also be in javadoc as well, given that the later will be easier to find long after the PR is merged ;)

Reply:
Added to PR description and javadoc. Should that metric be a broker-level gauge ?